gqcp/include/FockSpace/WaveFunction/WaveFunctionSelection.hpp

```diff
@@ -0,0 +1,75 @@
+// This file is part of GQCG-gqcp.
+// 
+// Copyright (C) 2017-2019  the GQCG developers
+// 
+// GQCG-gqcp is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// GQCG-gqcp is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-gqcp.  If not, see <http://www.gnu.org/licenses/>.
+// 
+#pragma once
+
+
+#include "Basis/TransformationMatrix.hpp"
+#include "FockSpace/SelectedFockSpace.hpp"
+#include "FockSpace/WaveFunction/WaveFunction.hpp"
+#include "Mathematical/Representation/Matrix.hpp"
+
+
+
+namespace GQCP {
+
+
+/**
+ *  A class that represents a selected wave function: (potentially non normalized) expansion coefficients in a Fock space
+ */
+class WaveFunctionSelection {
```

Comment:
Could you provide your rationale as to why this class exists?

How is it different from constructing a new wave function instance (e.g. with API `wave_function.select(...)`) whose `SelectedFockSpace` is smaller?

Reply:
The coefficients aren't normalized, it can be a slice from a wavefunction

Reply:
Could you mention this in the documentation?

Maybe something in the lines of "A subselection of a SelectedWaveFunction."